### PR TITLE
Bug 1144848 - Adding activity handling for addon details. r=eragonj

### PR DESCRIPTION
**Pull request by crh0716**
https://bugzilla.mozilla.org/show_bug.cgi?id=1144848

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=14d0d52d356a4c25d93e07567a3283eabbdb2f21

**Comment by EragonJ** [comment]
It would be nice if we can add comment for return values here : 

https://github.com/mozilla-b2g/gaia/pull/29475/files#diff-d6baeb121b6fb6283ffb5fd2e4a664d4L47

File: apps/settings/js/panels/addon_details/panel.js [hunk 1]
```diff
@@ -31,14 +31,23 @@ define(function(require) {
       },
 
       onBeforeShow: function(panel, options) {
-        this._curAddon = options.addon;
+        return Promise.resolve((() => {
+          if (options.addon) {
+            return options.addon;
+          } else if (options.manifestURL) {
+            return AddonManager.findAddonByManifestURL(options.manifestURL);
+          }
```
Comment (EragonJ) [inline apps/settings/js/panels/addon_details/panel.js]: Why we have to explicitly wrap this into a function ? Can't we just make sure what the app is at first and directly use that ?

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=22a2312567e615b9a0e089bc154c1ce584cd23f3

File: apps/settings/js/panels/addon_details/panel.js (hunk 1 above)
Comment (crh0716) [inline apps/settings/js/panels/addon_details/panel.js]: hmm, just trying to make it fit to the promise chain...

**Pull request merged**